app/waf/test/acceptance/application.controller.acceptance.ts

```diff
@@ -144,27 +133,30 @@ describe('ApplicationController', () => {
   });
 
   it(
-    'post ' +
-      prefix +
-      '/applications/{id}/deploy: no ADC associated with application',
+    'post ' + prefix + '/applications/{id}/deploy: deploy application',
     async () => {
-      const application = await givenApplicationData(wafapp);
+      let adc = await givenAdcData(wafapp);
+      let application = await givenApplicationData(wafapp, {
+        adcId: adc.id,
+      });
+
+      await client
+        .post(prefix + '/applications/' + application.id + '/declarations')
+        .send({name: 'a-declaration'})
+        .expect(200);
+
+      deployStub.returns(Promise.resolve('Hello'));
```

Comment:
The deloyStub makes the acceptance not as pure as we expected. 

Along time, there's more and more API call to AS3, I prefer to start up a rest server to mock the response outside, just like mock-<openstack> does. 
It's up to you.